src/core/runner.vala

```diff
@@ -0,0 +1,13 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private interface Games.Runner : Object {
+	public abstract Gtk.Widget? get_display ();
```

Comment:
> I don't really like having a domain class depend on UI.

It's true that it's suboptimal, but at least we have something to improve on. =)

> And aren't emulated games the only ones that we are able to run in a Gtk.Widget right now?
> Does it even make sense for other (Desktop/Wine/Steam/Web) games?

For now yes, I have crazy dreams of overtaking a game's window to display it in the application, doing Wayland or X dark magic or something.

Web games could be integrated using WebkitGTK. =)

> And what if I still want to run my SNES games with say ZSNES?

Then use ZSNES and not GNOME Games : it's out of the application's focus.

> And allowing some classes to return null means that, every place where we call this method, we'll have to check whether the result is null.

We need a good way to discriminate the "embedded" runners from the "remote" ones. Or the "headed" runners from the "headless" ones, I don't know how to call them yet. Do you have any suggestion?